### PR TITLE
Rebuild mod_custom with new Joomla 4 structure

### PR DESCRIPTION
**Pull request by sergeytolkachyov**
### Summary of Changes
Module Custom HTML has rebuilded with new Joomla 4 structure

### Testing Instructions
Just try to use it as usual.

### Actual result BEFORE applying this Pull Request
Module works

### Expected result AFTER applying this Pull Request
Module works

### Link to documentations
Please select:
- [X] Documentation link for docs.joomla.org: <link>
- [X] No documentation changes for docs.joomla.org needed

- [X] Pull Request link for manual.joomla.org: <link>
- [X] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: modules/mod_custom/src/Dispatcher/Dispatcher.php [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+     * @return  array
+     *
+     * @since   4.0.0
+     */
+    protected function getLayoutData()
+    {
+        $data = parent::getLayoutData();
+
+        if (($data['params'])->get('prepare_content', 1)) {
+            PluginHelper::importPlugin('content');
```
Comment (ReLater) [inline modules/mod_custom/src/Dispatcher/Dispatcher.php]:
```suggestion
```

HTMLHelper content.prepare imports the content plugins. Therefore I think not needed here.

File: modules/mod_custom/src/Dispatcher/Dispatcher.php [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+namespace Joomla\Module\Custom\Site\Dispatcher;
+
+// phpcs:disable PSR1.Files.SideEffects
+\defined('JPATH_PLATFORM') or die;
+// phpcs:enable PSR1.Files.SideEffects
+
+use Joomla\CMS\Dispatcher\AbstractModuleDispatcher;
+use Joomla\CMS\HTML\HTMLHelper;
+use Joomla\CMS\Plugin\PluginHelper;
```
Comment (ReLater) [inline modules/mod_custom/src/Dispatcher/Dispatcher.php]:
```suggestion
```

HTMLHelper content.prepare imports the content plugins. Therefore I think not needed here. See also comment below.

File: modules/mod_custom/tmpl/default.php [hunk 3]
```diff
@@ -4,15 +4,32 @@
...
+ * @copyright   (C) 2023 Open Source Matters, Inc. <https://www.joomla.org>
  * @license     GNU General Public License version 2 or later; see LICENSE.txt
  */
 
-defined('_JEXEC') or die;
+// phpcs:disable PSR1.Files.SideEffects
+\defined('JPATH_PLATFORM') or die;
+// phpcs:enable PSR1.Files.SideEffects
 
+use Joomla\CMS\Application\CMSApplicationInterface;
```
Comment (ReLater) [inline modules/mod_custom/tmpl/default.php]: Where used below?

File: modules/mod_custom/tmpl/default.php [hunk 4]
```diff
@@ -4,15 +4,32 @@
...
 use Joomla\CMS\Factory;
-use Joomla\CMS\HTML\HTMLHelper;
+use Joomla\Input\Input;
```
Comment (ReLater) [inline modules/mod_custom/tmpl/default.php]: Where used below?

File: modules/mod_custom/tmpl/default.php [hunk 5]
```diff
@@ -4,15 +4,32 @@
...
+use Joomla\Registry\Registry;
```
Comment (ReLater) [inline modules/mod_custom/tmpl/default.php]: Where used below?

File: modules/mod_custom/tmpl/default.php (hunk 3 above)
Comment (sergeytolkachyov) [reply modules/mod_custom/tmpl/default.php]: PHP Storm inserted it for `@var .....` line for reference purposes. We can remove it.

File: modules/mod_custom/tmpl/default.php (hunk 5 above)
Comment (sergeytolkachyov) [reply modules/mod_custom/tmpl/default.php:18]: See line 40. It is used for `HTMLHelper::_('cleanImageURL', $params->get('backgroundimage'))->url`

File: modules/mod_custom/src/Dispatcher/Dispatcher.php (hunk 2 above)
Comment (sergeytolkachyov) [reply modules/mod_custom/src/Dispatcher/Dispatcher.php:18]: We need to prepare content with content plugins somewhere. We can move this to module layout, but I think it isn't a good idea.

**Comment by richard67** [comment]
This should be done for the 4.3-dev branch.

**Comment by richard67** [comment]
> > This should be done for the 4.3-dev branch.
> 
> What should I do? Close this PR and pull again into 4.3-dev?

This, or rebase it to 4.3-dev by editing on GitHub, whatever is more comfortable to you.

**Comment by sergeytolkachyov** [comment]
> This, or rebase it to 4.3-dev by editing on GitHub, whatever is more comfortable to you.
@richard67 done

**Comment by richard67** [comment]
@sergeytolkachyov Thank you.

**Pull request closed without merging**